src/main/java/it/pagopa/transactions/commands/handlers/TransactionAddUserReceiptHandler.java

```diff
@@ -112,7 +127,39 @@ public Mono<TransactionUserReceiptAddedEvent> handle(TransactionAddUserReceiptCo
                                 }
                             });
 
-                    return emailResponse.flatMap(v -> transactionEventStoreRepository.save(event));
+                    return emailResponse
+                            .flatMap(v -> userReceiptAddedEventRepository.save(event))
+                            .flatMap(e -> {
+                                if (command.getData().addUserReceiptRequest()
+                                        .getOutcome() == AddUserReceiptRequestDto.OutcomeEnum.KO) {
+                                    TransactionExpiredEvent expirationEvent = new TransactionExpiredEvent(
+                                            transactionId,
+                                            new TransactionExpiredData(TransactionStatusDto.NOTIFIED_KO)
+                                    );
+
+                                    TransactionRefundRequestedEvent refundRequestedEvent = new TransactionRefundRequestedEvent(
+                                            transactionId,
+                                            new TransactionRefundedData(TransactionStatusDto.EXPIRED)
+                                    );
+
+                                    return expiredEventStoreRepository.save(expirationEvent)
+                                            .then(
```

Comment:
Remove this TransactionExpiredEvent, in this case the transactions can go directly in refund requested event and then the event can be written to the refund queue